File: 18_Frequency_Domain_Acoustics/Panel_Method_Matlab_Code/Panel Method/Panel_Method_Matlab_Code/panel_geometry.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%                                                                      %
%  panel_geometry.m - Compute airfoil surface panelization parameters  %
%                      for later use in the computation of the matrix  %
%                      of influence coefficients.                      %
%                                                                      %
%                                                                      %
%  Input list:                                                         %
%                                                                      %
%  x       -  Vector of x coordinates of the surface nodes             %
%  y       -  Vector of y coordinates of the surface nodes             %
%  npanel  -  Number of panels on the airfoil                          %
%                                                                      %
%  Output list:                                                        %
%                                                                      %
%  l       -  Panel lenghts                                            %
%  st      -  Sin(theta) for each panel                                %
%  ct      -  Cos(theta) for each panel                                %
%  xbar    -  X-coordinate of the midpoint of each panel               %
%  ybar    -  X-coordinate of the midpoint of each panel               %
%                                                                      %
%  Written by: Matthew Clarke                                          %
%              Department of Aerospace Engineering                     %
%              University of Illinois, Urbana-Champaign                % 
%              maclarke@illinois.edu                                   %
%                                                                      %
%  Last Modified: Wed July 2023                                        %
%                                                                      %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


function [l,st,ct,xbar,ybar] = panel_geometry(x,y,npanel);

%
% compute various geometrical quantities
%

for i=1:npanel
    l   (i) = sqrt((x(i+1) -x(i))^2 +(y(i+1) -y(i))^2);
    st  (i) = (y(i+1) -y(i))/l(i);
    ct  (i) = (x(i+1) -x(i))/l(i);
    xbar(i) = (x(i+1) +x(i))/2;
    ybar(i) = (y(i+1) +y(i))/2;
end

return

File: Panel Method/Panel_Method_Python_Code/panel_geometry.py
########################################################################
#                                                                      #
#  panel_geometry.py - Compute airfoil surface panelization parameters #
#                      for later use in the computation of the matrix  #
#                      of influence coefficients.                      #
#                                                                      #
#                                                                      #
#  Input list:                                                         #
#                                                                      #
#  x       -  Vector of x coordinates of the surface nodes             #
#  y       -  Vector of y coordinates of the surface nodes             #
#  npanel  -  Number of panels on the airfoil                          #
#                                                                      #
#  Output list:                                                        #
#                                                                      #
#  l       -  Panel lenghts                                            #
#  sin_theta      -  Sin(theta) for each panel                                #
#  cos_theta      -  Cos(theta) for each panel                                #
#  xbar    -  X-coordinate of the midpoint of each panel               #
#  ybar    -  X-coordinate of the midpoint of each panel               #
#                                                                      #
#  Written by: Matthew Clarke                                          #
#              Department of Aerospace Engineering                     #
#              University of Illinois, Urbana-Champaign                # 
#              maclarke@illinois.edu                                   #
#                                                                      #
#  Last Modified: Wed Aug 30 2023                                      #
#                                                                      #
########################################################################

import numpy as np 

def panel_geometry(x, y, npanel):
    # ---------------------------------------------------------------------------  
    # STEP 3.1 allocate all necessary arrays
    # ---------------------------------------------------------------------------  
    l           = np.zeros(npanel)
    sin_theta   = np.zeros(npanel)
    cos_theta   = np.zeros(npanel)
    xbar        = np.zeros(npanel)
    ybar        = np.zeros(npanel)

    # ---------------------------------------------------------------------------  
    # STEP 3.2 compute various geometrical quantities
    # ---------------------------------------------------------------------------  
    for i in range(npanel):
        l[i]    = np.sqrt((x[i + 1] - x[i]) ** 2 + (y[i + 1] - y[i]) ** 2)
        sin_theta[i]   = (y[i + 1] - y[i]) / l[i]
        cos_theta[i]   = (x[i + 1] - x[i]) / l[i]
        xbar[i] = (x[i + 1] + x[i]) / 2
        ybar[i] = (y[i + 1] + y[i]) / 2
     
    return l, sin_theta, cos_theta, xbar, ybar
